pkg/apis/kops/validation/validation_test.go

```diff
@@ -460,6 +461,7 @@ func Test_Validate_AdditionalPolicies(t *testing.T) {
 			Networking: kops.NetworkingSpec{
 				NetworkCIDR:           "10.10.0.0/16",
 				NonMasqueradeCIDR:     "100.64.0.0/10",
+				PodCIDR:               "100.96.0.0/11",
```

Comment:
Shouldn't `PodCIDR` and `ServiceClusterIPRange` be both part of `NonMasqueradeCIDR`?

Reply:
ServiceClusterIPRange doesn't have to be part of NonMasqueradeCIDR. That is because only things within the cluster network can talk to ServiceClusterIPRange and kube-proxy reroutes ServiceClusterIPRange before masquerading. GCE currently doesn't require ServiceClusterIPRange to be in NonMasqueradeCIDR; this PR drops that validation for all of the other cloud providers as well.

Reply:
In this case, does is the NonMasqueradeCIDR still used for anything? It was dropped by kubelet when Docker shim was removed.

Reply:
It's used by ContainerdBuilder and AWSCloudControllerManagerOptionsBuilder. It is also used as an exclusion for the egress proxy.

Reply:
The use by AWSCloudControllerManagerOptionsBuilder looks to me like it's a bug. It should be using the PodCIDR instead.

Reply:
Maybe we should use PodCIDR instead:
```
eccm.ClusterCIDR = clusterSpec.Networking.NonMasqueradeCIDR
```

Reply:
We can handle AWSCloudControllerManagerOptionsBuilder as a separate PR. That one we might want to backport.